### PR TITLE
chore: add gatsby-source-nacelle package

### PR DESCRIPTION
**Pull request by jasonredick**
### WHY are these changes introduced?

Story [[LS-1486](https://nacelle.atlassian.net/browse/LS-1486)] 

### WHAT is this pull request doing?

Adds `gatsby-source-nacelle` to `packages`. Removed from `getnacelle/nacelle-react` in PR: https://github.com/getnacelle/nacelle-react/pull/79

### How to Test

1. Run `npm run bootstrap`
2. Run `npm run test:ci` to ensure tests run with no errors.
3. Run `npm run build` to ensure packages build with no errors.

### CODE REVIEW

**Review by NWRichmond — APPROVED** [review]
The `bootstrap` works as expected:

LS-1486--move-gatsby-source-nacelle bootstrap

and all of the files removed in https://github.com/getnacelle/nacelle-react/pull/79/files are present and accounted for here.

Nice! Thanks for this PR, @jasonredick!

File: packages/gatsby-source-nacelle/package.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    "eslint": "^7.6.0",
+    "lint-staged": "^10.2.11"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/getnacelle/nacelle-js.git"
+  },
+  "bugs": {
+    "url": "https://github.com/getnacelle/nacelle-js/issues"
+  },
```
Comment (NWRichmond) [inline packages/gatsby-source-nacelle/package.json:48]: Excellent!

File: packages/gatsby-source-nacelle/package.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    "**/*.(js|ts|tsx)": [
+      "prettier --write",
+      "eslint",
+      "git add"
+    ],
+    "*{json, md}": [
+      "prettier --write"
+    ]
+  },
+  "gitHead": "3153c14f529fa9f691e7cd2e6c44cee74d1a78db"
```
Comment (NWRichmond) [inline packages/gatsby-source-nacelle/package.json:62]: I think we should remove the `gitHead` here, as it refers to an unrelated history.

```suggestion
```

**Review by NWRichmond — CHANGES_REQUESTED** [review]
Sorry @jasonredick I just noticed this one potential problem:

File: packages/gatsby-source-nacelle/package.json (hunk 2 above)
Comment (jasonredick) [reply packages/gatsby-source-nacelle/package.json]: @NWRichmond, removed!

Comment (NWRichmond) [reply packages/gatsby-source-nacelle/package.json]: Thank you!

**Review by NWRichmond — APPROVED** [review]
Great! Thank you for that update, @jasonredick.

**Pull request merged**